### ISSUE TITLE
"ionic serve --lab" not correctly updating the code

### ISSUE THREAD

**Issue by Zerokk** [ionic-team/ionic-cli#2590]
Quick bug report: 

It's been working until just a few hours ago, without having made any change to anything in the PC, NPM, Node... now I just have to use the command "ionic serve --lab" everytime I want it to proper update the files.

Otherwise, the command seems to respond when I save something from Visual Studio Code, like if it worked. But it just doesn't, since the code acts exactly the same as it did before. When I stop the command and run it again, it properly packs the code and executes it.

Why does this happen?

**Comment by dwieeb** [comment]
@Zerokk Please fill out the issue template.

Also, are there any console errors?

**Comment by Zerokk** [comment]
I guess the important info is this ionic info output:

> cli packages:
> 
>     @ionic/cli-plugin-cordova       : 1.5.0 (C:\Dev\Projects\StepApp\StepApp\node_modules\@ionic\cli-plugin-cordova)
>     @ionic/cli-plugin-ionic-angular : 1.4.0 (C:\Dev\Projects\StepApp\StepApp\node_modules\@ionic\cli-plugin-ionic-angular)
>     @ionic/cli-utils                : 1.6.0 (C:\Dev\Projects\StepApp\StepApp\node_modules\@ionic\cli-utils)
>     ionic (Ionic CLI)               : 3.6.0 (C:\Dev\Projects\StepApp\StepApp\node_modules\ionic)
> 
> global packages:
> 
>     Cordova CLI : 7.0.0
> 
> local packages:
> 
>     @ionic/app-scripts : 1.3.7
>     Cordova Platforms  : android 6.2.3
>     Ionic Framework    : ionic-angular 3.4.2
> 
> System:
> 
>     Node : v7.9.0
>     OS   : Windows 10
>     npm  : 4.2.0

@dwieeb No, console doesn't throw any errors, just hints (like unused classes in some parts of the project). In fact, the web view reloads fine and some changes are done, but not every change is done. Since two days ago, I normally have to re-run the command if I want to have the exact code I written, being executed.

**Comment by dwieeb** [comment]
I have been unable to reproduce, FYI.

Please try updating `@ionic/app-scripts` to the latest version.

**Comment by Zerokk** [comment]
I've updated app-scripts, yet I keep having the same problem. Sometimes,
some of the code changes, sometimes not. What I know this far, is that if I
don't re-run the script each time I change something in my code, the app
just won't work as intended.

Mind that this is happening in a Windows environment.

2017-08-02 17:23 GMT+02:00 dwieeb <notifications@github.com>:

**Comment by dwieeb** [comment]
@Zerokk I am testing in a Windows environment. I have edited TS, HTML, and SCSS files and all is working well.

Can you be more specific re: "Sometimes, some of the code changes, sometimes not"?

**Comment by Zerokk** [comment]
Sure.

I experience this daily. I work on a fairly big Ionic 3 project, at least
for what I use to do (over 35 components). I work using Visual Studio Code,
and when I save changes, the ionic serve --lab cmd console, instantly
starts to update the files. It's like if it works good.

Thing is, when the app loads, some of the changes I made, do not appear.
Some, do. It's completely crazy, but normally I have to re-launch the
command if I want to get the precise results of what I just did.

Per example, imagine that I comment a line where I get a null error, like:

    myObject.get("property");


After commenting the line and saving the file, I will (because I normally
do) get the same error I was having before. Just as if the line wasn't
commented.

That's just a typical problem I'm having that gets solved just re-launching
the command. But as this command takes a long time to work, this is so
uncomfortable.

2017-08-08 4:29 GMT+02:00 dwieeb <notifications@github.com>:
